src/QuickMarc.js

```diff
@@ -1,31 +1,36 @@
-import React from 'react';
+import React, { useCallback } from 'react';
 import PropTypes from 'prop-types';
 import {
   Switch,
   Route,
 } from 'react-router-dom';
 
-const QuickMarcEditor = () => (
-  <div>
-    QuickMarc
-  </div>
-);
+import { QuickMarcEditorContainer } from './QuickMarcEditor';
+
+const QuickMarc = ({ basePath, onClose }) => {
+  const QuickMarcEdit = useCallback(() => {
+    return (
+      <QuickMarcEditorContainer
+        onClose={onClose}
+      />
+    );
+  }, [onClose]);
 
-const QuickMarc = ({ baseRoute }) => {
   return (
     <div data-test-quick-marc>
       <Switch>
         <Route
-          component={QuickMarcEditor}
-          path={`${baseRoute}/quick-marc`}
+          path={`${basePath}/edit/:instanceId`}
+          component={QuickMarcEdit}
```

Comment:
is it legal to use as a component result of `useCallback` - regular function?

Reply:
function component is function, `QuickMarcEdit` is functional component here, but wrapped with`useCallback`

Reply:
yes, but what the reason? performance? it's unusual, IMO

Reply:
it's just a replacement of `render`, and `useCallback` is used to have the same link, nothing much